students/piechowski_michal/lesson_12_debugging/validated_user_base.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+import argparse
+import csv
+import logging
+import sys
+
+from students.piechowski_michal.lesson_08_regular_expressions.email_validator\
+    import is_valid as is_valid_email
+from students.piechowski_michal.lesson_08_regular_expressions.phone_number_validator\
+    import is_valid as is_valid_phone_number
+from students.piechowski_michal.lesson_08_regular_expressions.postal_code_validator\
+    import is_valid as is_valid_postal_code
+from students.piechowski_michal.lesson_08_regular_expressions.strong_password_detection\
+    import is_strong as is_strong_password
+
+logging.basicConfig(level=logging.INFO,
+                    format=' %(asctime)s - %(levelname)s - %(message)s')
+
+
+def combined_validator(email, password, phone_number, postal_code):
+    logging.info("Verifying user data")
+
+    if is_valid_email(email) and is_strong_password(password) \
+            and is_valid_phone_number(phone_number) \
+            and is_valid_postal_code(postal_code):
+        logging.info("Verification passed")
+    else:
+        logging.warning("Verification failed")
+        raise Exception("One of arguments didn't pass verification.")
+
+
+def check_arg(args):
+    parser = argparse.ArgumentParser(
+        description='Script to save validated users to a CSV file')
+    parser.add_argument('-l', '--log',
+                        choices=['disabled', 'warning', 'info'],
+                        help='Log verbosity',
+                        default='info')
+
+    results = parser.parse_args(args)
+
+    if results.log == 'disabled':
+        return logging.CRITICAL
+    elif results.log == 'warning':
+        return logging.WARNING
+    else:
+        return logging.INFO
+
+
+def save_user_data(email, password, phone_number, postal_code):
+    csv_file = 'userdata.csv'
+
+    # create file if it doesn't exist
+    with open(csv_file, 'a') as file:
+        pass
+
+    # check if user is already in the file and if yes remove it
+    with open(csv_file, 'r') as file:
```

Comment:
ogólnie może być. Chociaz mozna by użyc tymczasowego pliku i wykonywac odczyt z `csv_file` i jednoczesnie zapis do `tmp_csv_file` i jeżeli trafiliśmy na nasz email to podmienić dane, a jesli go nie było w pliku to dopisac do `tmp_csv_file` i po zamknięciu plikiem tmp zastapic wejściowy (funkcja `shutil.move`)